packages/plugins/minos-router-graphql/minos/plugins/graphql/handlers.py

```diff
@@ -0,0 +1,56 @@
+from graphql import (
+    GraphQLSchema,
+    graphql,
+    print_schema,
+)
+
+from minos.networks import (
+    HttpRequest,
+    HttpResponse,
+    ResponseException,
+)
+
+
+class GraphQlHandler:
+    """GraphQl Handler"""
+
+    def __init__(self, schema: GraphQLSchema):
+        self._schema = schema
+
+    async def execute_operation(self, request: HttpRequest) -> HttpResponse:
+        """Execute incoming request extracting variables and passing to graphql"""
+
+        content = await request.content()
+
+        source = dict()
+        variables = dict()
+
+        if isinstance(content, str):
+            source = content
+
+        if isinstance(content, dict):
+            if "query" in content:
+                source = content["query"]
+
+            if "variables" in content:
+                variables = content["variables"]
+
+        result = await graphql(schema=self._schema, source=source, variable_values=variables)
+
+        errors = result.errors
+        if errors is None:
+            errors = list()
+
+        status = 200
+
+        if len(errors):
+            status = 400
```

Comment:
The default `status` should be `500`, only overwritten by a `ResponseException` status 🙂